Lib/fontbakery/utils.py

```diff
@@ -537,19 +538,19 @@ def iterate_lookup_list_with_extensions(ttFont, table, callback, *args):
     if table not in ttFont or not ttFont[table].table.LookupList:
         return
 
+    # This function mutates the TTF; deep copy to avoid this, and so avoid
+    # issues with concurrent tests that also use ttFont.
+    # See https://github.com/fonttools/fontbakery/issues/4834
+    ttFont = deepcopy(ttFont)
```

Comment:
My understanding is that Python gives function parameters their own independent variable to store their reference to an object, but agree that it's 100% worth double-checking; I had a look just now to be safe, and it looks like we're all good 🎉 

```python
from dataclasses import dataclass

@dataclass
class Wrapper:
    value: int

def replacer(parameter: Wrapper) -> None:
    parameter = Wrapper(value=2)

outside = Wrapper(value=1)
print(outside)  # prints Wrapper(value=1)
replacer(outside)
print(outside)  # prints Wrapper(value=1) ✅
```

I considered giving the variable a new name, but thought that reusing it would avoid extra refactoring and would have the bonus of making it impossible for the rest of the function to use the pre-copied object by mistake.

Reply:
Sorry, I've just pushed a few commits changing the names, but I had not seen these messages before doing so. I think I'll revert my additional commit, then.

 > Failing pytest appears to be unrelated;

Just to be sure we're on the same page, what was the pytest failure message in this case?

Reply:
No worries, thanks for taking a look Felipe! 🚀

Yes, reverting sounds good - the test that I am seeing ERROR locally is `unreachable_glyphs`, which appears to be [the one upsetting the CI too](https://github.com/fonttools/fontbakery/actions/runs/10890521212/job/30219819275#step:6:951), with full error:

```
AttributeError: 'TTFont' object has no attribute 'glyphOrder'. Did you mean: 'getGlyphOrder'?
```

I thought it was unrelated before, but now I see that it is one of the checks that we have touched in this PR.

I think what might be happening is that the check previously populated `.glyphOrder` as a side-effect, but now that we deepcopy it is not happening any more; this is probably a good thing, so I think the pytest just needs to be updated to call `.getGlyphOrder()` instead of trying to access the raw property to accommodate this :)

Reply:
It turned out to be a little more complicated because the test relied on certain context to remove a glyph; I think I have a full fix with fe7ab5e0dd1ed562d45ad1a82419da8f63840dcc though (fuller explanation there too)